CONTRIBUTING.md

```diff
@@ -29,3 +29,142 @@ Recommendations and requirements for how to best contribute to Parquet. We striv
+   some information might be missing or performance might be suboptimal.
+
+3. Forward incompatible. A file written under a newer version of the format with
+   the feature enabled cannot be read under an older version of the format (e.g.
+   adding and using a new compression algorithm). It is expected any feature in
+   this category will provide a signal to older readers, so they can
+   unambiguously determine that they cannot properly read the file (e.g. via
+   adding a new value to an existing enum).
+
+New features are intended to be widely beneficial to users of Parquet, and
+therefore it is hoped third-party implementations will adopt them quickly after
+they are introduced. It is assumed that writing new parts of the format, and
+especially forward incompatible features, will be configured with a feature flag
+defaulted to "off", and at some future point the feature is turned on by default
+(reading of the new feature will typically be enabled without configuration or
+defaulted to on). Some amount of lead time is desirable to ensure a critical
+mass of Parquet implementations support a feature to avoid compatibility issues
+across the ecosystem.  Therefore, the Parquet PMC gives the following
+recommendations for managing features:
+
+1. Backward compatibility is the concern of implementations but given the
+   ubiquity of Parquet and the length of time it has been used, libraries should
+   support reading older versions of the format to the greatest extent possible.
+
+2. Forward compatible features/changes may be enabled and used by default in
+   implementations once the parquet-format containing those changes has been
+   formally released.  For features that may pose a significant performance
+   regression to older format readers, libaries should consider delaying default
+   enablement until 1 year after the release of the parquet-java implementation
+   that contains the feature implementation.
+
+3. Forward incompatible features/changes should not be turned on by default
+   until 2 years after the parquet-java implementation containing the feature is
+   released. It is recommended that changing the default value for a forward
+   incompatible feature flag should be clearly advertised to consumers (e.g. via
+   a major version release if using Semantic Versioning, or highlighed in
+   release notes).
+
+For forward compatible changes which have a high chance of performance
+regression for older readers and forward incompatible changes, implementations
+should clearly document the compatibility issues. Additionally, while it is up
+to maintainers of individual implementations to make the best decision to serve
+their ecosystem, they are encouraged to start enabling features by default along
+the same timelines as `parquet-java`.  Parquet-java will wait to enable features
+by default until the most conservative timelines outlined above have been
+exceeded.
```

Comment:
I think they should be encourage implementations to enable it as soon as they are comfortable with it and no later than the library unless they have legacy constraints.
The conservative approach in parquet-java is because it has much less visible potential impact.

Using the compatibility matrix to turn things on by default more aggressively when something is quickly adopted is a great idea in my opinion

Reply:
> I think they should be encourage implementations to enable it as soon as they are comfortable with it and no later than the library unless they have legacy constraints.

Is the library here `parquet-java`.  

> Using the compatibility matrix to turn things on by default more aggressively when something is quickly adopted is a great idea in my opinion

I agree for early adopters it is good, but these are most likely people who understand there dependency stack very well.  I think the issue is even with release dates, these libraries take time to percolate through the ecosystem so being more conservative with language here makes sense to me.  

If you can think of language that might strike a better balance here, I am open to suggestions.

Reply:
> > I think they should be encourage implementations to enable it as soon as they are comfortable with it and no later than the library unless they have legacy constraints.

> Is the library here parquet-java.

Here, I meant the library they're using. So either parquet-java, parquet-rust or parquet-cpp depending on the context.

> I agree for early adopters it is good, but these are most likely people who understand there dependency stack very well. I think the issue is even with release dates, these libraries take time to percolate through the ecosystem so being more conservative with language here makes sense to me.
> If you can think of language that might strike a better balance here, I am open to suggestions.

(Here, I'm considering that most third party implementation are actually integrated in a broader system or query engine)

How about: 

_Additionally, while it is up to maintainers of individual implementations to make the best decision to serve
their ecosystem, they are encouraged to start enabling features by default as soon as potential consumers of the files they produce will be able to read them. They can follow the same timelines as `parquet-java` but should consider that it follows a conservative timeline._

Is this pushing too much the other way?

Reply:
I am addding language to clarify the guidance is for open source implementations and then adding a sentence:

```
We encourage earlier adoption of new features when 
an organization using Parquet can guarantee that
all readers of the parquet files they produce can read a new 
feature
```

I'll resolve this conversation when we push I push the commit and we can see if further editing is needed.

Reply:
pushed commit.  resolving.